### PR TITLE
Four sorries in CNodeInv_R.thy

### PR DESCRIPTION
**Pull request by MitchellBuckley**
This fixes 4 proofs in CNodeInv_R.thy

### CODE REVIEW

**Review by michaelmcinerney — APPROVED** [review]
Ahhh yeahhhh

**Review by lsf37 — APPROVED** [review]
👍

File: proof/refine/ARM/CNodeInv_R.thy [hunk 1]
```diff
@@ -5516,13 +5520,19 @@ lemma make_zombie_invs':
...
                                \<and> obj_at' (Not \<circ> tcbQueued) p' s
-                               \<and> bound_tcb_at' ((=) None) p' s")
+                               \<and> bound_tcb_at' ((=) None) p' s
+                               \<and> obj_at' (\<lambda>tcb. tcbSchedContext tcb = None
+                                   \<or> tcbSchedContext tcb = Some idle_sc_ptr) p' s
+                               \<and> obj_at' (\<lambda>tcb. tcbYieldTo tcb = None) p' s")
     apply (clarsimp simp: pred_tcb_at'_def obj_at'_def ko_wp_at'_def projectKOs)
-   apply (auto dest!: isCapDs)[1]
-  sorry (*
+    apply (auto dest!: isCapDs)[1]
```
Comment (mbrcknl) [inline proof/refine/ARM/CNodeInv_R.thy:5532]: `subgoal by`? Also indentation looks a bit strange.

**Review by mbrcknl — APPROVED** [review]
Nice!

File: proof/refine/ARM/CNodeInv_R.thy [hunk 2]
```diff
@@ -5481,6 +5481,10 @@ lemma make_zombie_invs':
                       (\<forall>p \<in> threadCapRefs (cteCap cte).
                             st_tcb_at' ((=) Inactive) p s
                              \<and> bound_tcb_at' ((=) None) p s
+                             \<and> obj_at' (\<lambda>tcb. tcbSchedContext tcb = None \<or> tcbSchedContext tcb = Some idle_sc_ptr) p s
```
Comment (mktnk3) [inline proof/refine/ARM/CNodeInv_R.thy:5484]: We have `bound_sc_tcb_at’` and `bound_yt_tcb_at’` for these, just like `bound_tcb_at’` above.

**Review by mktnk3 — APPROVED** [review]

**Pull request merged**